ipfscluster_test.go

```diff
@@ -163,26 +166,41 @@ func createClusters(t *testing.T) ([]*Cluster, []*test.IpfsMock) {
 		clusterPeers[i] = addr
 	}
 
-	// Set up the cluster using ClusterPeers
-	for i := 0; i < nClusters; i++ {
-		cfgs[i].Peers = make([]ma.Multiaddr, nClusters, nClusters)
-		for j := 0; j < nClusters; j++ {
-			cfgs[i].Peers[j] = clusterPeers[j]
-		}
-	}
+	// ----------------------------------------------------------
 
-	// Alternative way of starting using bootstrap
-	// for i := 1; i < nClusters; i++ {
-	// 	addr, _ := ma.NewMultiaddr(fmt.Sprintf("/ip4/127.0.0.1/tcp/%d/ipfs/%s",
-	// 		clusterPort,
-	// 		cfgs[0].ID.Pretty()))
+	// // Set up the cluster using ClusterPeers
+	// for i := 0; i < nClusters; i++ {
+	// 	cfgs[i].Peers = make([]ma.Multiaddr, nClusters, nClusters)
+	// 	for j := 0; j < nClusters; j++ {
+	// 		cfgs[i].Peers[j] = clusterPeers[j]
+	// 	}
+	// }
 
-	// 	// Use previous cluster  for bootstrapping
-	// 	cfgs[i].Bootstrap = []ma.Multiaddr{addr}
+	// var wg sync.WaitGroup
+	// for i := 0; i < nClusters; i++ {
+	// 	wg.Add(1)
+	// 	go func(i int) {
+	// 		clusters[i] = createCluster(t, cfgs[i], concfgs[i], apis[i], ipfss[i], states[i], trackers[i], mons[i], allocs[i], infs[i])
+	// 		wg.Done()
+	// 	}(i)
 	// }
+	// wg.Wait()
+
+	// ----------------------------------------------
 
+	// Alternative way of starting using bootstrap
+	// Start first node
+	clusters[0] = createCluster(t, cfgs[0], concfgs[0], apis[0], ipfss[0], states[0], trackers[0], mons[0], allocs[0], infs[0])
+	// Find out where it binded
+	bootstrapAddr, _ := ma.NewMultiaddr(fmt.Sprintf("%s/ipfs/%s", clusters[0].host.Addrs()[0], clusters[0].id.Pretty()))
+	// Use first node to bootstrap
+	for i := 1; i < nClusters; i++ {
```

Comment:
Is the bootstrap method required when moving to using port 0 for some reason?  Again this is just curiosity.

Reply:
yeah, cannot use the `Peers` array in the configuration because we have no idea where they are listening on before they are started. So need to start one peer and use it as bootstrap for the rest, in order to form the cluster. same as in kubernetes btw.